AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainSystem_VegetationSpawnsOnTerrainSurfaces.py

```diff
@@ -0,0 +1,217 @@
+    vegetation_entity.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    # Set the vegetation area to a Dynamic Slice spawner with a specific slice asset selected.
+    dynamic_slice_spawner = vegetation.DynamicSliceInstanceSpawner()
+    dynamic_slice_spawner.SetSliceAssetPath(os.path.join("Slices", "PinkFlower.dynamicslice"))
+    descriptor = hydra.get_component_property_value(vegetation_entity.components[2], 'Configuration|Embedded Assets|[0]')
+    descriptor.spawner = dynamic_slice_spawner
+    vegetation_entity.get_set_test(2, "Configuration|Embedded Assets|[0]", descriptor)
+
+    # Assign gradients to layer spawners.
+    terrain_entity_1.get_set_test(2, "Configuration|Gradient Entities", [gradient_entity_1.id])
+    terrain_entity_2.get_set_test(2, "Configuration|Gradient Entities", [gradient_entity_2.id])
+
+    # Move view so that the entitiess are visible.
+    general.set_current_view_position(17.0, -66.0, 41.0)
+    general.set_current_view_rotation(-15, 0, 0)
+
+    # Wait for the vegetation to spawn
+    helper.wait_for_condition(lambda: vegetation.VegetationSpawnerRequestBus(bus.Event, "GetAreaProductCount", vegetation_entity.id) > 300, 5.0)
+
+    # Check the spawn count is correct.
+    item_count = vegetation.VegetationSpawnerRequestBus(bus.Event, "GetAreaProductCount", vegetation_entity.id)
+    Report.result(VegetationTests.unfiltered_vegetation_count_correct, item_count > 300)
+
+    test_aabb = math.Aabb_CreateFromMinMax(math.Vector3(-10.0, -10.0, 0.0), math.Vector3(30.0, 10.0, box_height))
+
+    # Find the z positions of the items with the lowest and highest x values, this will avoid the overlap area where z values are blended between the surface heights.
+    highest_z, lowest_z = FindHighestAndLowestZValuesInArea(test_aabb)
+    
+    # Check that the z values are as expected.
+    Report.result(VegetationTests.vegetation_on_gradient_1, lowest_z == box_height * gradient_value_1)
+    Report.result(VegetationTests.vegetation_on_gradient_2, highest_z == box_height * gradient_value_2)
+
+    # Assign SurfaceTags to the SurfaceGradientLists
+    terrain_entity_1.get_set_test(3, "Configuration|Gradient to Surface Mappings|[0]|Surface Tag|Surface Tag", surface_tags["test_tag2"])
+    terrain_entity_2.get_set_test(3, "Configuration|Gradient to Surface Mappings|[0]|Surface Tag|Surface Tag", surface_tags["test_tag3"])
+
+    # Give the VegetationSurfaceFilter an exclusion list, set it to excude test_tag2 which should remove all the lower items which are in terrain_entity_1.
+    tag_list = [surface_data.SurfaceTag()]
+
+    vegetation_entity.get_set_test(3, "Configuration|Exclusion|Surface Tags", tag_list)
+    vegetation_entity.get_set_test(3, "Configuration|Exclusion|Surface Tags|[0]|Surface Tag", surface_tags["test_tag2"])
```

Comment:
Suggestion:  Instead of tag_list, you might just be able to use [].  Not sure you need to give it an entry with an unassigned tag.

Reply:
No, it needs an item in the list unfortunately. I've removed the variable holding it though.